deps/chakrashim/lib/chakra_debug.js

```diff
@@ -0,0 +1,1304 @@
+// Copyright Microsoft. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the 'Software'), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and / or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+
+/* eslint-disable strict */
+(function(globalObject, keepAliveObject, traceDebugJson) {
+  var globalExecutionState;
+  var pendingMessages = [];
+
+  var Logger = (function() {
+    var logTypes = {
+      API: 1,
+      Output: 2,
+      Input: 3,
+      Error: 4
+    };
+
+    var enabledLogType = [];
+    enabledLogType[logTypes.API] = false; // For development purpose
+    enabledLogType[logTypes.Output] = traceDebugJson === true;
+    enabledLogType[logTypes.Input] = traceDebugJson === true;
+    enabledLogType[logTypes.Error] = false; // For development purpose
```

Comment:
Now we only have LogAPI, DebugJson and Error. DebugJson is enabled using --trace_debug_json and Error is enabled by default.

Seems like when we execute chakra_debug process object is not yet set so can't use that to get the environment variable. Will see if I can do something for dynamically enable/disable after the PR as a separate change.

---
In reply to: [82242030](https://github.com/nodejs/node-chakracore/pull/126#discussion_r82242030) [](ancestors = 82242030)